modules/node-group/main.tf

```
@@ -27,8 +29,15 @@ locals {
     },
     "storage" : merge(
       var.use_privnet ? local.privnet_config : {},
-      local.is_storage_cluster ? local.storage_cluster_config : {},
+      // We create a custom partition layout if data_disk_size or
+      // storage_disk_size are > 0. This is primarily to avoid issues with sector
+      // rounding during partitioning if the user didn't request extra disk space.
+      var.data_disk_size + var.storage_disk_size > 0 ? local.disks_config : {},
     ),
+    // For storage cluster nodes we zero the extra partition on first boot to
+    // ensure deploying the storage cluster succeeds.
+    // We don't do this for other nodes where users request extra disk space via
+    // data_disk_size.
```

What would you say, we also follow the one-sentence-per-line styling?
Would make the comment more maintainable (and IMO, more readable as well)
```suggestion
      // We create a custom partition layout if data_disk_size or storage_disk_size are > 0. 
      // This is primarily to avoid issues with sector rounding during partitioning if the user didn't request extra disk space.
      var.data_disk_size + var.storage_disk_size > 0 ? local.disks_config : {},
    ),
    // For storage cluster nodes we zero the extra partition on first boot to ensure deploying the storage cluster succeeds.
    // We don't do this for other nodes where users request extra disk space via data_disk_size.
```

---

Personally, I find long comment lines much harder to read than nicely wrapped ones.